kotest-assertions/src/commonMain/kotlin/io/kotest/assertions/show/DefaultShow.kt

```diff
@@ -55,3 +57,7 @@ private fun Collection<*>.getCollectionSnippet(): Printed {
 internal fun recursiveRepr(root: Any, node: Any?): Printed {
    return if (root == node) "(this ${root::class.simpleName})".printed() else node.show()
 }
+
+private fun handleRecursiveIterable(i: Iterable<*>, kClass: KClass<out Any>) : Any {
+   return if (i.iterator().hasNext() && kClass.isInstance(i.iterator().next())) i.toString() else i.toList()
```

Comment:
Can you add a comment to this function explaining how it works because it's not clear to me.

Reply:
I've been thinking over it a little bit more and recognized that this solution won't work properly :(